src/vscode-dts/vscode.proposed.chatSlashCommands.d.ts

```diff
@@ -0,0 +1,43 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	export interface SlashCommandContext {
+
+		// messages so far
+		history: ChatMessage[];
+
+		// TODO: access to embeddings
+		// embeddings: {};
+
+		// TODO: access to "InputSourceId"
+		// DebugConsoleOutput
+		// Terminal
+		// CorrespondingTestFile
+		// CorrespondingImplementationFile
+		// ExtensionApi
+		// VSCode
+		// Workspace
+	}
+
+	export interface SlashResponse {
+		message: MarkdownString;
+		// edits?: TextEdit[] | WorkspaceEdit;
+	}
+
+	export interface SlashResult {
+		// followUp?: InteractiveSessionFollowup[];
+	}
+
+	export interface SlashCommand {
```

Comment:
Will at least need a description for the slash command. There are also some other metadata fields that Joyce added https://github.com/microsoft/vscode/blob/af667558efb055a047080722900c500107a4f7c5/src/vscode-dts/vscode.proposed.interactive.d.ts#L155-L158

Reply:
Actually I guess I'm not sure if this is expected to replace the old `provideSlashCommands` on the chat provider or if you are _only_ thinking about contributions from other extensions here

Reply:
For now it's just extension contributed commands but we can extend this as we go. The description/detail property exists but as of today it is defined statically inside package.json. This needs to change for dynamic commands, is actually already implemented like this but I forgot expose it here